fullcalendar-app/src/components/CalendarHeader.vue

```diff
@@ -0,0 +1,188 @@
+<template>
+  <div class="calendar-branch-header">
+    <div class="calendar-branch-info" v-if="branch">
+      <h4>{{ 'Branch name' }}</h4>
+      <h2>{{ branch }}</h2>
+    </div>
+  </div>
+
+  <div class="fullcalendar--header">
+    <div class="pdf">
+      <button class="download-pdf-button" @click="$emit('downloadPDF')">
+        <i class="bi bi-download"></i>
+        Download PDF
+      </button>
+    </div>
+
+    <button class="legend-toggle-button btn btn-dark" data-mdb-ripple-init @click="toggleLegend">Legend +</button>
+    <div :class="['checkbox-container', { active: isLegendOpen }]">
+      <label class="custom-checkbox" v-for="category in categories" :key="category.name">
+        <input
+          type="checkbox"
+          :value="category.name"
+          :checked="isSelected(category.name)"
+          @change="emitCategoryChange(category.name, $event.target.checked)"
+          class="custom-checkbox-input"
+        />
+        <span class="checkmark" :style="{ 'background-color': category.color, 'border-color': category.color }"></span>
+        {{ category.name }}
+      </label>
+    </div>
+  </div>
+</template>
+
+<script>
+import EventService from '../service/EventService';
+
+export default {
+  name: 'CalendarHeader',
+  props: {
+    categories: Array,
+    selectedCategories: Array
+  },
+  data() {
+    return {
+      isLegendOpen: false,
+      branch: null,
+    };
+  },
+  mounted() {
+    const eventService = new EventService();
+    this.branch = eventService.getBranch();
+  },
+  methods: {
+    toggleLegend() {
+      this.isLegendOpen = !this.isLegendOpen;
+    },
+    isSelected(categoryName) {
+      // Make sure selectedCategories is an array before calling includes
+      if (Array.isArray(this.selectedCategories)) {
+        return this.selectedCategories.includes(categoryName);
+      }
+      console.error('selectedCategories is not an array:', this.selectedCategories);
```

Comment:
Fixed.
Now it's something like
`      console.error('Error: selectedCategories is expected to be an array but received a different type.', {
        actionNeeded: 'Check the initialization of selectedCategories to ensure it is correctly set as an array. This might involve reviewing the data source or state management logic that provides value to selectedCategories.',
        receivedType: typeof this.selectedCategories,
        receivedValue: this.selectedCategories,
        suggestion: 'If selectedCategories data is dynamically loaded, verify the data structure returned by the server or external data source. Ensure any transformation or assignment operation maintains the array data type.',
      });`